### PR TITLE
Add support for websocket connections

### PR DESCRIPTION
**Pull request by mzanetti**
This commit adds support for connecting via the websocket transport
protocol on both, client and server parts.
In addition, the standalone server implemention is more complete, allowing
to manage policies and server ports via command line arguments as well
as adding a command line client tool for subscribing to a broker and/or
publishing messages to it.

### CODE REVIEW

**Comment by t-mon** [comment]
certificateloader.cpp:30:10: fatal error: openssl/ssl.h: No such file or directory
 #include <openssl/ssl.h>
          ^~~~~~~~~~~~~~~
compilation terminated.


Disable for now.

File: client/client.pro [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+QT += network
+QT -= gui
+
+INCLUDEPATH += $$top_srcdir/libnymea-mqtt/
+
+SOURCES += main.cpp
+
+LIBS += -L$${top_builddir}/libnymea-mqtt -lnymea-mqtt
+
+target.path = /usr/bin/
```
Comment (t-mon) [inline client/client.pro]: `$$[QT_INSTALL_PREFIX]/bin`

File: client/main.cpp [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+    if (parser.positionalArguments().count() < 1) {
+        parser.showHelp(-1);
+    }
+
+    QString verbose = parser.isSet("verbose") ? "true" : "false";
+    QLoggingCategory::setFilterRules(QString("nymea.mqtt.client.debug=%1\nnymea.mqtt.client.warning=%2\n").arg(verbose).arg(verbose));
+
+    bool ok;
+    Mqtt::QoS qos = static_cast<Mqtt::QoS>(parser.value("qos").toInt(&ok));
```
Comment (t-mon) [inline client/main.cpp]: you verify the convert, but not the range

File: client/main.cpp [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+                return;
+            }
+        }
+        qWarning() << "SSL errors for certificate:" << qUtf8Printable(certificate.toText()) << sslErrors;
+    });
+    QObject::connect(&client, &MqttClient::connected, app, [&parser, &client, qos, retain](Mqtt::ConnectReturnCode connectReturnCode, Mqtt::ConnackFlags /*connackFlags*/){
+        switch (connectReturnCode) {
+        case Mqtt::ConnectReturnCodeIdentifierRejected:
+            qCritical() << "Connection failed: Identifier rejected";
+            exit(-1);
```
Comment (t-mon) [inline client/main.cpp]: Not so important, only fyi: we actually have a constant for this called `EXIT_FAILURE` from stdlib return 1 and not -1.

File: libnymea-mqtt/transports/mqttclienttransport.cpp [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+*
+* For any further details and any questions please contact us under contact@nymea.io
+* or see our FAQ/Licensing Information on https://nymea.io/license/faq
+*
+* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#include "mqttclienttransport.h"
+
+MqttClientTransport::MqttClientTransport(QObject *parent)
+    : QObject{parent}
```
Comment (t-mon) [inline libnymea-mqtt/transports/mqttclienttransport.cpp]: `QObject(parent)`? ...not sure what about the `{` `}`

**Review by t-mon — APPROVED** [review]
nice

File: client/client.pro (hunk 1 above)
Comment (mzanetti) [reply client/client.pro]: fixed, thanks

File: client/main.cpp (hunk 2 above)
Comment (mzanetti) [reply client/main.cpp]: fixed

File: client/main.cpp (hunk 3 above)
Comment (mzanetti) [reply client/main.cpp]: fixed

File: libnymea-mqtt/transports/mqttclienttransport.cpp (hunk 4 above)
Comment (mzanetti) [reply libnymea-mqtt/transports/mqttclienttransport.cpp]: Ah, newest QtCreator does this by default... As far as I can tell, the only difference is that with () the variables are initialized in the order they are written here, while using {} they are initialized in the order they are declared in the .h.

**Review by t-mon — APPROVED** [review]

**Pull request merged**